clash-ghc/src-ghc/Clash/GHC/Evaluator/Primitive.hs

```diff
@@ -5884,5 +5937,5 @@ svoid :: (State# RealWorld -> State# RealWorld) -> IO ()
 svoid m0 = IO (\s -> case m0 s of s' -> (# s', () #))
 
 isTrueDC,isFalseDC :: DataCon -> Bool
-isTrueDC dc  = dcUniq dc == getKey trueDataConKey
-isFalseDC dc = dcUniq dc == getKey falseDataConKey
+isTrueDC dc  = dcUniq dc == fromIntegral (getKey trueDataConKey)
+isFalseDC dc = dcUniq dc == fromIntegral (getKey falseDataConKey)
```

Comment:
Instead of sprinkling around a very generic function like `fromIntegral`, I feel it might smart to define a conversion function from GHC uniques to clash uniques and use that everywhere.

Reply:
That function could for now just be `fromIntegral`, but if we need to change it in the future, we have a central place to do so.

Reply:
I'll make a function called `fromGhcUnique`.